SingularityUI/app/collections/TaskSearchResults.coffee

```diff
@@ -0,0 +1,63 @@
+Collection = require './collection'
+
+TaskSearchResult = require '../models/TaskSearchResult'
+
+class TaskSearchResults extends Collection
+
+    model: TaskSearchResult
+
+    params: ->
+    	anyParams = false
+    	base = '?'
+    	if @deployId
+    		base += 'deployId=' + @deployId
+    		anyParams = true
+    	if @host
+    		if anyParams
+    			base += '&'
+    		base += 'host=' + @host
+    		anyParams = true
+    	if @lastTaskStatus
+    		if anyParams
+    			base += '&'
+    		base += 'lastTaskStatus=' + @lastTaskStatus
+    		anyParams = true
+    	if @startedAfter
+    		if anyParams
+    			base += '&'
+    		base += 'startedAfter=' + @startedAfter
+    		anyParams = true
+    	if @startedBefore
+    		if anyParams
+    			base += '&'
+    		base += 'startedBefore=' + @startedBefore
+    		anyParams = true
+    	if @orderDirection
+    		if anyParams
+    			base += '&'
+    		base += 'orderDirection=' + @orderDirection
+    		anyParams = true
+    	if @count
+    		if anyParams
+    			base += '&'
+    		base += 'count=' + @count
+    		anyParams = true
+    	if @page
+    		if anyParams
+    			base += '&'
+    		base += 'page=' + @page
+    		anyParams = true
+    	return '' if not anyParams
+    	return base
```

Comment:
We can build the params a bit cleaner here. Probably easier to build up an array of params (i.e. if @attribute then add to array), then return paramsArray.join('&')